### PR TITLE
Run type checks from root

### LINKED ISSUE #383: Improve type checking performance
Now that we are using TypeScript project references, we can improve the type checking performance by calling `tsc --build` from the root and adding the references from `tsconfig.build.json` to the root `tsconfig.json`. This is ~4x faster (CPU time) than calling `tsc --build` in each package using lerna due to the tsc/lerna overhead.

Remove the types scripts in each package and have the root level scripts call `tsc --build` or `tsc --build --watch` when necessary. This can be done async w/ the other build steps since babel doesn't care about type checking and just strips TS notation.

Also check how many file watchers this affects. It should reduce file watchers compared to running tsc in each package.

### PR DESCRIPTION
**Pull request by mattrunyon**
Closes #383 

This runs type checks from the root of the repo to improve performance.

Locally went from ~300s CPU time for a build to ~200s.

It also should improve the start time of `npm start` since the type checking is cached

### CODE REVIEW

**Review by mofojed — CHANGES_REQUESTED** [review]

**Comment by mofojed** [comment]
I'm getting an error when I run `npm start`:

```
@deephaven/code-studio: ℹ ｢wds｣: Project is running at http://192.168.86.167/
@deephaven/code-studio: ℹ ｢wds｣: webpack output is served from 
@deephaven/code-studio: ℹ ｢wds｣: Content not from webpack is served from /home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/code-studio/public
@deephaven/code-studio: ℹ ｢wds｣: 404s will fallback to /
@deephaven/code-studio: Starting the development server...
[10:59:23 a.m.] Found 0 errors. Watching for file changes.

@deephaven/code-studio: Failed to compile.
@deephaven/code-studio: /home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/code-studio/src/log/LogExport.ts
@deephaven/code-studio: TypeScript error in /home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/code-studio/src/log/LogExport.ts(3,16):
@deephaven/code-studio: Output file '/home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/jsapi-shim/dist/index.d.ts' has not been built from source file '/home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/jsapi-shim/src/index.ts'.  TS6305
@deephaven/code-studio:     1 | /* eslint-disable import/prefer-default-export */
@deephaven/code-studio:     2 | import JSZip from 'jszip';
@deephaven/code-studio:   > 3 | import dh from '@deephaven/jsapi-shim';
@deephaven/code-studio:       |                ^
@deephaven/code-studio:     4 | import { store } from '@deephaven/redux';
@deephaven/code-studio:     5 | import { logHistory } from './LogInit';
@deephaven/code-studio:     6 |

```

File: packages/chart/src/ChartModelFactory.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ChartModelFactory {
    * @param {string[]} settings.hiddenSeries Array of hidden series names
    * @param {dh.Table} table The table to build the model for
    * @param {Object} theme The theme for the figure. Defaults to ChartTheme
-   * @returns {Promise<FigureChartModel>} The FigureChartModel representing the figure
+   * @returns {Promise<ChartModel | FigureChartModel>} The FigureChartModel representing the figure
```
[the remark this answers is not in the document]
Comment (mattrunyon) [reply packages/chart/src/ChartModelFactory.js:19]: Added comment about CRA type checking JS

File: packages/code-studio/src/main/WidgetUtils.ts [hunk 2]
```diff
@@ -54,7 +54,8 @@ export const createChartModel = async (
     const definition = { name: figureName, type: dh.VariableType.FIGURE };
     const figure = await session.getObject(definition);
 
-    return ChartModelFactory.makeModel(settings, figure);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return ChartModelFactory.makeModel(settings as any, figure);
```
[the remark this answers is not in the document]
Comment (mattrunyon) [reply packages/code-studio/src/main/WidgetUtils.ts:58]: Ya it was the easiest fix for now. W/ our tsconfig CRA was just ignoring type checks for a while

File: examples/component-template/package.json [hunk 3]
```diff
@@ -15,20 +15,19 @@
...
     "build:babel": "npm run babel",
     "build:sass": "npm run sass",
-    "build:types": "npm run types -- --incremental",
     "watch": "run-p watch:*",
     "watch:babel": "npm run babel -- -w --skip-initial-build",
     "watch:sass": "npm run sass -- --watch --update",
-    "watch:types": "npm run types -- -w --incremental --preserveWatchOutput",
-    "clean": "rimraf ./dist tsconfig.tsbuildinfo",
+    "clean": "run-p clean:*",
+    "clean:types": "tsc -b --clean",
```
[the remark this answers is not in the document]
Comment (mattrunyon) [reply examples/component-template/package.json:29]: Ya it's only an option in `build` mode. It determines what files would be created by `tsc --build` including the buildinfo cache file and then deletes them all

**Review by mofojed — APPROVED** [review]

**Pull request merged**